_posts/2016/2016-05-26-magdeburger-developer-days.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "Magdeburger Developer Days"
+date: "2016-05-26 12:30:20"
+image: blog-header/magdeburg-dev.jpg
+categories: events
+authors: ["Bastian and Benjamin"]
+---
+
+This May the Developer Days took place at the exhibition hall of Magdeburg, Germany. Although it was the first time many developers from nearly each part of Germany participated. There were also a lot of interesting talks with a broad range of topics.
+
+{% image blog/blog-magdeburg-1.jpg %}
+
+## The setting
+
+The exhibition hall was parted into four rooms, named after famous pioneers of the Information Technologies, and a lobby with bar tables, drinks and snacks.
+
+{% image blog/blog-magdeburg-2.jpg %}
+
+## The sessions
+
+After the two Introduction talks, the Sessions began. The main topics were Front-End technologies, Back-End technologies and a mixture of Scrum, Documentation and programming technologies. As there were always four talks in parallel I can only tell about the talks I visited.
+
+{% image blog/blog-magdeburg-3.jpg %}
+
+The first Session was an introduction into TypeScript by Rainer Stropek. TypeScript is a JavaScript similar language and can be compiled into JavaScript. It is a strictly typed and object-oriented programming language. And that is what this session was about. We learned how to compile and use TypeScript. Rainer Stropek showed examples live at the TypeScript Playground(https://www.typescriptlang.org/play/index.html), which directly generates the resulting JavaScript. He also showed how to use common JavaScript libraries within TypeScript.
+
+{% image blog/blog-magdeburg-4.jpg %}
+
+Our second session was all about "What's still new in Java SE8". Bernd Gronostay talked about a lot of new features within Java 8. But he did not only talk about them, but also showed a lot of examples which illustrated these features very well. The perhaps most interesting part in this session was the introductino into lambda functions and streams. This enables developers to write Java in a functional syntax.
+
+{% image blog/blog-magdeburg-5.jpg %}
+
+After lunch there was a talk with a topic which was completely new to me: Functional Domain Driven Design (DDD) held by Marco Heimeshoff. He is one of the cofounders of DDD. His introduction showed a new idea of how the model should look like. The mapping between the Application Space (Domain) and the Implementation Space (Code) is very hard. The difficulty is to ave the code representing exactly the real Domain. But often developers have to check that there is no possibility to represent an illegal state. The simple solution with DDD is: Make illegal states unrepresentable!
```

Comment:
After lunch there was a talk with a topic which was completely new to me:
> After lunch, there was a talk with a topic, which was completely new to me:

 The difficulty is to ave the code representing exactly the real Domain.
>  The difficulty is to have the code representing exactly the real domain.

But often developers have to check that there is no possibility to represent an illegal state.
> What do you mean by "have to check"? ...müssen erkennen = have to realise

The mapping between the Application Space (Domain) and the Implementation Space (Code) is very hard.
> The mapping between the Application Space (domain) and the Implementation Space (code) is very hard.